### PR TITLE
Ticket 3928: Update ICP during deploy

### PR DESCRIPTION
**Pull request by DominicOram**
See https://github.com/ISISComputingGroup/IBEX/issues/3928

To test:
1. Start your instrument
1. Create the folder `C:\Labview Modules\DAE` and unzip [icp_config.zip](https://github.com/ISISComputingGroup/ibex_utils/files/4912788/icp_config.zip) into it
1. Console into `ISISDAE`, toggle autorestart to off and kill the IOC (keep the console open)
1. In taskmanager kill `ISISICP` and confirm it doesn't restart itself (keep task manager open)
1. Run the upgrade script with `python IBEX_upgrade.py --kits_icp_dir \\isis.cclrc.ac.uk\inst$\Kits$\CompGroup\ICP\ --confirm_step instrument_deploy_main`
1. Reply `N` to the first 6 steps then `Y` to the update ICP step
1. Confirm the script copies from the DAE2 directory
1. After copy (and after you've entered admin credentials for the register) start the `ISISDAE` console
1. In task manager rightclick on ISISICP and look at the properties, confirm it is being run from inside the `Labview Modules` directory
1. Modify `icp_config.xml` to have DAEType of 3, delete the contents of `C:\Labview Modules\DAE\service`, repeat steps 2-8 but instead confirming that the copy happens from the DAE3 directory
1. Rename `C:\Labview Modules\DAE\` and repeat steps 2-8 but confirm that a new ICP is copied into `C:\Instrument\EPICS\ICP_binaries` and it is running from there

### CODE REVIEW

File: installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py [hunk 1]
```diff
@@ -327,20 +327,43 @@ def _pretty_print(data):
                     print("Couldn't get data from {} because: {}".format(pv, e.message))
                     f.write(e.message)
 
-    @task("Patch ISISDAE for ticket 5164")
```
Comment (Tom-Willemsen) [inline installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py:330]: This seems to have been lost and will now deploy the latest - was that intentional? As far as I was aware we couldn't use the latest isisdae ioc on MUONFE and had to patch back to 5.5.0 each time. 

If that's no longer true could you add a quick comment on https://github.com/ISISComputingGroup/IBEX/issues/5164 saying the patch is no longer required?

Comment (DominicOram) [reply installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py:330]: Yes, sorry, this is still required, not sure why I thought it wasn't

File: installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py [hunk 2]
```diff
@@ -327,20 +327,43 @@ def _pretty_print(data):
...
+            elif DAE_type in [3, 4]:
+                DAE_type = 3
+            else:
+                print("DAE type {} not recognised, not installing ICP".format(DAE_type))
+                return
+            self.prompt.confirm_step("Upgrade DAE{} type ICP found in Labview Modules".format(DAE_type))
+            ICP_path = get_latest_directory_path(os.path.join(INST_SHARE_AREA, "kits$", "CompGroup", "ICP", "ISISICP",
+                                                              "DAE{}".format(DAE_type)), "")
+
+            RunProcess(os.getcwd(), "update_inst.cmd", executable_directory=ICP_path).run()
```
Comment (FreddieAkeroyd) [inline installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py:354]: update_inst now always does a batch "pause" at the end unless you pass NOINT  as the first argument.

**Comment by DominicOram** [comment]
The register should now work. As long as you have the changes specified in https://github.com/ISISNeutronMuon/ISISICP/pull/25. If the ISISICP PR isn't merged when reviewing this you can manually copy the changes into `Labview Modules/DAE/register_programs.cmd` when you are prompted to give your admin credentials. Alternatively, the registering will still work without the ISISICP changes but will hang straight after being registered.

**Pull request merged**